src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemAlignmentUtils.java

```diff
@@ -91,11 +91,43 @@ else if ( alwaysGenerateASandXS ) {
     }
 
     /**
+     * Produces an SA tag for each primary line and supplemental alignment as a map from alignment to tag value.
+     */
+    public static Map<BwaMemAlignment,String> createSATags( final List<BwaMemAlignment> alignments,
+                                                            final List<String> refNames ) {
+        final int nAlignments = alignments.size();
+        if ( nAlignments < 2 ) return Collections.emptyMap();
+
+        final String[] selfTags = new String[nAlignments];
+        for ( int idx = 0; idx != nAlignments; ++idx ) {
+            final BwaMemAlignment alignment = alignments.get(idx);
+            if ( SAMFlag.NOT_PRIMARY_ALIGNMENT.isUnset(alignment.getSamFlag()) ) {
+                selfTags[idx] = asTag(alignment, refNames);
+            }
+        }
+
+        final Map<BwaMemAlignment,String> saTags = new HashMap<>(SVUtils.hashMapCapacity(nAlignments));
+        for ( int idx = 0; idx != nAlignments; ++idx ) {
+            final BwaMemAlignment alignment = alignments.get(idx);
+            if ( SAMFlag.NOT_PRIMARY_ALIGNMENT.isSet(alignment.getSamFlag()) ) continue;
+
+            final StringBuilder saTag = new StringBuilder();
+            for ( int idx2 = 0; idx2 != nAlignments; ++idx2 ) {
+                if ( idx2 == idx ) continue;
+                final String tag = selfTags[idx2];
+                if ( tag != null ) saTag.append(tag);
+            }
+            saTags.put(alignment, saTag.toString());
+        }
+        return saTags;
+    }
+
+    /**
      * Describes an alignment as a string for use in an SA tag, for example.
      */
     public static String asTag( final BwaMemAlignment alignment, final List<String> refNames ) {
```

Comment:
Yeah.  I don't see a need to make this private.  It's generally useful, even if it's only used to make SA tags right now.  The encoding used for SA and XA tags allows you to reference other alignments within an SAM record, and might see other uses.  So asTag pretty well describes turning an alignment into this kind of string, I think.
I think I'll duck your suggestion, if that's OK.